worker/restore.go

```diff
@@ -92,8 +94,160 @@ func RunRestore(pdir, location, backupId string, key x.SensitiveByteSlice,
 			if err != nil {
 				return 0, 0, err
 			}
+			isOld = in.isOld
 			return maxUid, maxNsId, x.WriteGroupIdFile(dir, uint32(groupId))
 		})
+
+	if result.Err == nil && isOld {
+		// Only group 1 contains the internal predicate.
+		dir := filepath.Join(pdir, fmt.Sprintf("p%d", 1))
+		pstore, result.Err = badger.OpenManaged(badger.DefaultOptions(dir).
+			WithCompression(ctype).
+			WithZSTDCompressionLevel(clevel).
+			WithSyncWrites(false).
+			WithBlockCacheSize(100 * (1 << 20)).
+			WithIndexCacheSize(100 * (1 << 20)).
+			WithNumVersionsToKeep(math.MaxInt32).
+			WithEncryptionKey(key).
+			WithNamespaceOffset(x.NamespaceOffset))
+		if result.Err != nil {
+			return result
+		}
+		defer pstore.Close()
+		result.Err = fixBackup()
+	}
+	return result
+}
+
+func getData(attr string, fn func(item *badger.Item) error) error {
+	return pstore.View(func(txn *badger.Txn) error {
+		attr = x.GalaxyAttr(attr)
+		initKey := x.ParsedKey{
+			Attr: attr,
+		}
+		prefix := initKey.DataPrefix()
+		startKey := append(x.DataKey(attr, math.MaxUint64))
+
+		itOpt := badger.DefaultIteratorOptions
+		itOpt.AllVersions = true
+		itOpt.Reverse = true
+		itOpt.Prefix = prefix
+		itr := txn.NewIterator(itOpt)
+		defer itr.Close()
+		for itr.Seek(startKey); itr.Valid(); itr.Next() {
+			item := itr.Item()
+			// We expect only complete posting list.
+			x.AssertTrue(item.UserMeta() == posting.BitCompletePosting)
+			if err := fn(item); err != nil {
+				return err
+			}
+			break
```

Comment:
You would always read a single key. 
If you just want to read a single key, run `txn.Get(...)` instead of creating the iterator.

Reply:
We don't know the exact key (specifically the UID to generate the correct key).